### DISCUSSION TITLE
Need help to install a plugin

### DISCUSSION THREAD

**Discussion by netoxss** [netbox-community/netbox#17315] [Help Wanted!] [answered]
i got this message after trying to run upgrade.sh and now i cant install any plugin ou upgrade the netbox 
![image](https://github.com/user-attachments/assets/90a6b81d-4c69-4b2c-ba2d-95d766896589)
i have attachments installed, in the local_requirements and in the configuration.py

**Comment by xavbroadcast** [comment]
A few questions:
- Have you used the netbox attachments plugin in your install (was it working before you attempted the current upgrade)?
- Is this your first upgrade attempt after installing the attachments plugin?
- the netbox attachments plugin is declared in your current configuration.py file?
- what command did you run that produced the screenshot? Had you activated the virtual environment before attempting the upgrade?

**Reply by netoxss** [reply]
1- yes, its already installed before, and no problems with him
2- no, after every plugin config and installation i run the upgrade.sh
3- yes, and now i notice, after removing him from configuration.py, the erro message just change to others plugins name
4- sudo /opt/netbox/upgrade.sh
![image](https://github.com/user-attachments/assets/27ceb43c-8210-4c84-8734-16cf30eb3fd0)

**Reply by xavbroadcast** [reply]
`yes, and now i notice, after removing him from configuration.py, the erro message just change to others plugins name` - do you mean you no longer need the plugin ?
I'm pretty sure you need to check that the list of plugins in your `configuration.py` file matches those listed in the `local_requirements.txt` file otherwise you will have errors on mismatches.

**Reply by netoxss** [reply]
1 - no, im still needing, after the failed upgrade.sh the netbox service wont starts now
2 - the local_requirements are:
![image](https://github.com/user-attachments/assets/20de61a7-6378-43d7-b2eb-5750b804c965)
and list of plugins in configuration.py are: 
netbox_metatype_importer','netbox_topology_views','netbox_documents','netbox_lists','netbox_attachments','netbox_reorder_rack','slurpit_netbox','validity','nb_risk','netbox_diode_plugin','netbox_floorpan_plugin','netbox_qrcode', 

In my view they are all the same

**Reply by xavbroadcast** [reply]
Assuming the array in your configuration.py doesn't end with `...'netbox_floorpan_plugin','netbox_qrcode',` : is there a comma after the qrcode entry? There shouldn't be if that's the last plugin, the following character should be the `]` to close the array.

If it's not something like that, then I can't help and we'd need to wait for a netbox expert to chime in with further questions if necessary.

**Reply by netoxss** [reply]
theres a ] in the end of plugins section: 
![image](https://github.com/user-attachments/assets/2f8b1203-d221-47b0-8025-b4f5f9ff6ecd)
thats only a error of my copy and paste

I will analyze my configuration.py and try to figure out whats wrong, thx for the help

**Reply by xavbroadcast** [reply]
Only other thought: this is an upgrade to Netbox 4.0.0 ? If so, which version of Netbox did you attempt to upgrade from? It's a major change so there are some prerequisites and breaking changes. Maybe some of your plugins aren't supported in v4.0 ?

If you have a backup, you could try restoring that, disabling all your plugins (both in local_requirements and in configuration.py) and attempt the upgrade. You won't want that long term of course, but it would help identify if it's a plugin causing the problem with the new v4.0.

**Reply by netoxss** [reply]
nono, this a fresh installation directly in the v4.0.0, i will try to disable all plugins and enabling one per one

**Reply by netoxss** [reply]
> Only other thought: this is an upgrade to Netbox 4.0.0 ? If so, which version of Netbox did you attempt to upgrade from? It's a major change so there are some prerequisites and breaking changes. Maybe some of your plugins aren't supported in v4.0 ?
> 
> If you have a backup, you could try restoring that, disabling all your plugins (both in local_requirements and in configuration.py) and attempt the upgrade. You won't want that long term of course, but it would help identify if it's a plugin causing the problem with the new v4.0.

removed diode, floorplan and qr code from configuration and now its working, buts returned me an advise:

![image](https://github.com/user-attachments/assets/66f194e9-38f8-40b0-a40a-287e2912b6cb)

**Reply by xavbroadcast** [reply]
OK, and if you now, as root, activate your virtual environment, then run the `pip install netbox-attachments` then add the `netbox_attachments` to your plugin list then run `python3 manage.py migrate netbox_attachments` (basically following the install).

**Reply by netoxss** [reply, marked as answer]
![image](https://github.com/user-attachments/assets/6f56b4e3-2322-48b4-80bd-64b08c78d00d)
the problem was the flooplan plugin, they dont heave a v4.0.0 support

**Reply by netoxss** [reply]
> ![image](https://private-user-images.githubusercontent.com/120464223/363247723-6f56b4e3-2322-48b4-80bd-64b08c78d00d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMzQyNjEsIm5iZiI6MTcyNTAzMzk2MSwicGF0aCI6Ii8xMjA0NjQyMjMvMzYzMjQ3NzIzLTZmNTZiNGUzLTIzMjItNDhiNC04MGJkLTY0YjA4Yzc4ZDAwZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgzMFQxNjA2MDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05NDE2ZTQ4MWMxYTUxMzBlNTA5NzhlYzdkYmYyOTg4ZmYwYTcxOWEyNGY5MzBiNWU3ZmIxN2FmN2IzNjkwZWRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.TUrLxwkKSp7y1ch5d0Rf3TOJkdIq0FM3I9uICiJIUx0) the problem was the flooplan plugin, they dont heave a v4.0.0 support

after removing from configurations.py, it works fine

@xavbroadcast thanks for the help man, have a nice day

**Comment by candlerb** [comment]
The plugin supports >4.0.2, so you should install 4.0.10, not 4.0.0.

Point releases are primarily bug fixes, so by choosing 4.0.0 you're exposing yourself to lots of bugs that have already been found and fixed.

**Reply by netoxss** [reply]
i will check the other plugins compatibility with 4.0.10 and think about the upgrade, thanks for the suggestion
